docs/topic_guide/extending_fido.rst

```diff
@@ -27,29 +27,43 @@ A new "scraper" client inherits from `~sunpy.net.dataretriever.client.GenericCli
   Each `tuple` contains the "attr" value and its description.
-* A class attribute ``pattern``; this is a string used to match the URLs supported by the client and extract metadata from the matched URLs.
-  The time and other metadata attributes for extraction are written using the `~sunpy.extern.parse.parse` format, using double curly-brackets so to differentiate them from ``kwargs`` parameters which are written in single curly-brackets.
+* A class attribute ``pattern``; this is a string used to match all URLs supported by the client and extract necessary metadata from the matched URLs.
+  The time and other metadata attributes for extraction are written within double curly-braces ``{{}}`` in a `parse <https://github.com/r1chardj0n3s/parse/>`__ format.
+  Regular placeholders for Python format strings can still be included via single curly braces ``{}`` with their respective parameters passed as the ``kwargs``.
+  An example of how such a pattern looks like is given in the algorithm explanation below.
+
+Note: The scraper has supported regex-based patterns for a long time, which is what it will still expect by default. However that is in the process of being replaced with parse-style patterns in the future versions of the package. To use the newer parse-style patterns, it's currently required to pass it as the ``format`` argument to the scraper.
 
 Each such client relies on the `~sunpy.net.scraper.Scraper` to be able to query for files using the :meth:`~sunpy.net.scraper.Scraper.filelist` method. The general algorithm to explain how the `~sunpy.net.scraper.Scraper` is able to do this is:
+A brief explanation of how the Scraper works is as follows:
+
+1. Drop the filename from the pattern and generate a list of directories to search for files.
+2. For each directory, get a list of files present in it.
+3. For each file, check if it matches the pattern.
+4. For each file that matches the pattern, check if it is in the timerange. If it is, add it to the output.
+
+For a more verbose in-depth explanation on how this is accomplished internally:
 
-1. It takes as input a generalised ``pattern`` of how a desired filepath looks like, following the ``parse`` format. A version of the pattern following the datetime format is also generated, called the ``timepattern``.
+1. A Scraper object takes as input the generalised ``pattern`` of how a desired filepath looks like, in the ``parse`` format. Upon initialisation, a version of the pattern following the datetime format is also internally generated, called the ``dt_pattern``.
 
 .. code-block:: python
 
     >>> from sunpy.net import Scraper
     >>> pattern = ('http://proba2.oma.be/{instrument}/data/bsd/{{year:4d}}/{{month:2d}}/{{day:2d}}/'
     ...            '{instrument}_lv1_{{year:4d}}{{month:2d}}{{day:2d}}_{{hour:2d}}{{minute:2d}}{{second:2d}}.fits')
-    >>> s = Scraper(pattern, instrument='swap')
+    >>> s = Scraper(format=pattern, instrument='swap')
+    >>> s.dt_pattern
```

Comment:
Thoughts on renaming this to "datetiime_pattern"?

Reply:
i think that works better, yeah